### PR TITLE
httpd: Serve root handler on `/api/v1` instead of `/api`

### PR DESCRIPTION
**Pull request by xphoniex**
```console
$ curl localhost:8080/api | jq
{
  "path": "/api",
  "links": [
    "/v1"
  ]
}

$ curl localhost:8080/api/v1 | jq
{
  "message": "Welcome!",
  "service": "radicle-httpd",
  "version": "0.1.0-b2c2b65",
  "node": {
    "id": "z6MkqgfPWLDruEjqzY5enrrqSSSxpARdFsQK6DWJfBBV9AV4"
  },
  "path": "/api/v1",
  "links": [
    {
      "href": "/projects",
      "rel": "projects",
      "type": "GET"
    },
    {
      "href": "/node",
      "rel": "node",
      "type": "GET"
    },
    {
      "href": "/delegates/:did/projects",
      "rel": "projects",
      "type": "GET"
    },
    {
      "href": "/stats",
      "rel": "stats",
      "type": "GET"
    }
  ]
}
```

### CODE REVIEW

**Comment by rudolfs** [comment]
Just looking at the outputs you posted. Shouldn't _everything_ be under `/api/v1` including `message`, `service`, etc?

**Comment by xphoniex** [comment]
> Just looking at the outputs you posted. Shouldn't _everything_ be under `/api/v1` including `{"message":"Welcome!","service":"radicle-httpd","version":"0.1.0-b2c2b65","node":{"id":"z6MkwPUeUS2fJMfc2HZN1RQTQcTTuhw4HhPySB8JeUg2mVvx"},"path":"/api","links":[{"canonical":"/v1"}]}`?
> 
> If we still have stuff under plain `/api` then that doesn't fix anything in the client lib for me.

Thanks for clarifying this, I wanted to ask about this. Should I revert @cloudhead ?

**Comment by cloudhead** [comment]
Yeah, though both have merit, I would say that `/api` is not in itself an API and therefore should only provide the minimum links to the actual API, which is `/api/v1`.

**Comment by rudolfs** [comment]
The links seem to be outdated or incomplete btw. I guess we write them manually?

**Comment by xphoniex** [comment]
> The links seem to be outdated or incomplete btw. I guess we write them manually? 

Which other part is outdated/incomplete? `/sessions` is missing.

> Also, shouldn't the links be shown at the very top-level, i.e. `/`? People would have to guess that it's `/api` in the first place.

api simply starts from `/api`, and we shouldn't expect hitting `"/"` to give us api-related info, as that's not clearly api, and falls into git router area.

**Comment by rudolfs** [comment]
> > The links seem to be outdated or incomplete btw. I guess we write them manually?
> 
> Which other part is outdated/incomplete? `/sessions` is missing.

I guess the rest of the project routes:
```
/projects/:project
/projects/:project/commits
/projects/:project/commits/:sha
/projects/:project/activity
/projects/:project/tree/:sha
/projects/:project/tree/:sha/*path
/projects/:project/remotes
/projects/:project/remotes/:peer
/projects/:project/blob/:sha/*path
/projects/:project/readme/:sha
/projects/:project/issues
/projects/:project/issues/:id
/projects/:project/patches
/projects/:project/patches/:id
```

> > Also, shouldn't the links be shown at the very top-level, i.e. `/`? People would have to guess that it's `/api` in the first place.
> 
> api simply starts from `/api`, and we shouldn't expect hitting `"/"` to give us api-related info, as that's not clearly api, and falls into git router area.

ah, forgot that we serve git from it. 👍

**Comment by xphoniex** [comment]
I thought these were supposed to give high level view of routes, do we want that granularity @cloudhead ?

**Comment by xphoniex** [comment]
> Do we need the routes showing up there?

Probably not, unless interface needs it? @rudolfs 

This was copied from `http-api`.

File: radicle-httpd/src/api.rs [hunk 1]
```diff
@@ -98,35 +93,10 @@ pub fn router(ctx: Context) -> Router {
...
-                "type": "GET"
-            },
-            {
-                "href": "/v1/stats",
-                "rel": "stats",
-                "type": "GET"
-            }
-        ]
+        "path": "/api",
+        "links": ["/v1"]
```
Comment (cloudhead) [inline radicle-httpd/src/api.rs:99]: Maybe use the same format as the in /v1? Eg.  with href/rel/type?

**Comment by rudolfs** [comment]
> 

I don't need the routes.

**Pull request merged**